docs/usage.md

```diff
@@ -0,0 +1,323 @@
+                opentracing::NoopSpan,
+                opentracing::NoopOptions,
+                opentracing::NoopContext,
+                opentracing::NoopAdapter> Tracer;
+    // Tracer is required for O(1) implementation swaps
+
+    typedef Tracer::SpanContext SpanContext;
+    typedef Tracer::SpanOptions SpanOptions;
+    typedef Tracer::Span        Span;
+    // For convenience
+}
+#endif
+```
+
+By creating the `Tracer` typedef, we can now rely on the GenericTracer interface,
+but avoid knowing any of the explicit implementation details in client code.
+
+The `Tracer` exposes a number of public typedefs which make it easier to create our
+types:
+
+ - `Tracer::Span`             - Interface to Span objects
+ - `Tracer::SpanContext`      - Interface to SpanContext objects
+ - `Tracer::SpanOptions`      - Interface to SpanOptions objects
+
+Our theoretical "ACME Inc." organization can use these type names to instrument their applications.
+The `Tracer` typedef is not standardized, but it is recommended. The rest of the documentation
+will assume that `Tracer` was the chosen name for the Tracer type.
+
+If clients ever require that they swap `OpenTracing` implementations, they
+will need to update only this header file and rebuild applications.
+(Easier said then done for larger organizations...)
+
+### Installing the Tracer
+
+Once a implementation is chosen, your applications can rely on your global `opentracing-cpp`
+configuration header to set up the `Tracer`. The `Tracer` implementation singleton is installed
+explicitly in main. This must be done before any other parts of the API can be used:
+
+```
+// acmetask.m.cpp
+#include <acme_tracing.h>
+
+using namespace acme;
+
+int main(int argc, const char * argv[])
+{
+    TracerImpl tracerImpl;
+    Tracer::install(&tracerImpl);
+
+    // ...
+
+    return 0;
+}
+
+```
+
+### Creating Spans
+
+Once the `Tracer` is installed, we can begin to instrument the application.
+Say we have a service that acts as a proxy to a number of other HTTP servers.
+We may have a `getAccount` handler that reaches out to a user account service.
+
+To trace our application, we'll first need to start the span:
+
+```
+// requesthandler.m.cpp
+#include <acme_tracing.h>
+
+using namespace acme;
+
+int getAccount(Repsonse * resp, const Request& req)
+{
+    Tracer::Span* span(Tracer::start("get_account"));
+    assert(span);
+
+    // When our span is complete, we have to call finish
+    span->finish();
+
+    // When we're done with our Span, we need to return it to
+    // the Tracer implementation to be cleaned up
+    Tracer::cleanup(span);
+
+    return 0;
+}
+```
+
+### Defining carriers
+
+In order to instrument our entire network, we need a way to `inject` and `extract`
+spans into and out of our RPC calls. Extending our example, when we make our request
+to the backend HTTP service, we'll need a way to inject the details of the context
+into our outgoing HTTP request.
```

Comment:
I'd want to see the outcome of that HTTP issue before I started to implement the HTTP carriers, but the usage example could be cleaned up regardless.